### PR TITLE
Return 0.1 for the default bulb position - fixes a nearClip == 0 DIV/0 error

### PR DESCRIPTION
**Pull request by carb-lloyd**
When the general implement of GetBulbPositionOffset is called, the frustum's nearClip ends up as 0.0.  This causes a DIV/0 that finally causes an assert down the line about a non-normalized vector.

The fix here is just to return 0.01f as the default

### CODE REVIEW

**Comment by AMZN-alexpete** [comment]
@akioCL is it expected that the bulb position offset should always be non-zero? or should we handle a zero offset at the call sites?

**Comment by moudgils** [comment]
@akioCL to approve but I think you shouldn't have to add **#if defined(CARBONATED).** for this fix

File: Gems/Atom/Feature/Common/Code/Source/CoreLights/SpotLightUtils.h [hunk 1]
```diff
@@ -40,7 +40,11 @@ namespace AZ
             template<class LightData>
             float GetBulbPositionOffset([[maybe_unused]] const LightData& light, general_)
             {
+#if defined(CARBONATED)
+                return 0.01f;
+#else
                 return 0.0f;
+#endif
```
Comment (akioCL) [inline Gems/Atom/Feature/Common/Code/Source/CoreLights/SpotLightUtils.h]:
```suggestion
                return 0.01f;
```

Comment (akioCL) [reply Gems/Atom/Feature/Common/Code/Source/CoreLights/SpotLightUtils.h]: The fix is correct but there's no need to add the **#if defined(CARBONATED)**

**Review by moudgils — APPROVED** [review]

**Pull request merged**